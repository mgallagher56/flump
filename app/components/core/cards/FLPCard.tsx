import type { FC } from 'react';

import type { CardProps } from '@chakra-ui/react';
import { Card } from '@chakra-ui/react';

import { FLPCardStyles } from './FLPCardStyles';

const FLPCard: FC<CardProps> = (props) => {
  return (
    <Card className={FLPCardStyles} {...props}>
      {props.children}
    </Card>
  );
};
export default FLPCard;
