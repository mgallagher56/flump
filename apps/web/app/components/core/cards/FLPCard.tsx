import type { FC } from 'react';

import type { CardRootProps } from '@chakra-ui/react';
import { CardRoot } from '@chakra-ui/react';

import { FLPCardStyles } from './FLPCardStyles';

const FLPCard: FC<CardRootProps> = (props) => {
  return (
    <CardRoot className={FLPCardStyles} {...props}>
      {props.children}
    </CardRoot>
  );
};
export default FLPCard;
