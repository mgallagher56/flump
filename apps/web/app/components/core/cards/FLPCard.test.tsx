import { render } from '@testing-library/react';

import FLPCard from './FLPCard';

describe('<FlpCard />', () => {
  test('it renders a ChakraUI Card component with title as expected', () => {
    const { container } = render(<FLPCard title="FLPCard" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
