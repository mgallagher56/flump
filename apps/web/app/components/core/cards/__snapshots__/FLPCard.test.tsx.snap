// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<FlpCard /> > it renders a ChakraUI Card component with title as expected 1`] = `
<div
  class="chakra-card pos_relative trs_all_0.3s_ease-in-out bd_1px_solid_token(colors.gray.500)! [&:hover]:bx-sh_2px_5px_8px_0px_token(colors.gray.500)! [&:hover]:trf_translate3d(-2px,_-5px,_-5px) [&:hover]:cursor_pointer css-2dlz7"
  title="FLPCard"
/>
`;
